mongo/description/server.go

```diff
@@ -318,12 +315,7 @@ func NewServerFromError(addr address.Address, err error, tv *TopologyVersion) Se
 // SetAverageRTT sets the average round trip time for this server description.
 func (s Server) SetAverageRTT(rtt time.Duration) Server {
 	s.AverageRTT = rtt
-	if rtt == UnsetRTT {
-		s.AverageRTTSet = false
-	} else {
-		s.AverageRTTSet = true
-	}
-
+	s.AverageRTTSet = true
```

Comment:
A little tangential, but looking at the callers, one thing I was unsure of was whether https://github.com/mongodb/mongo-go-driver/blob/a325a9904efe52455073c65a009b2d3504dca7d8/x/mongo/driver/topology/server.go#L715 should check if the RTT monitor's `averageRTTSet` was set before overwriting the server descriptions. I think that would never happen though, since when opening a connection in server monitoring, the RTT is set with the handshake. When the RTT monitor is reset on error, the monitoring connection is closed.

Reply:
I think it's an invariant that the `rttMonitor` has at least one sample by the time we get to this line. As you mentioned, we must've opened a monitoring connection and doing so adds the sample from the connection handshake. Also, errors in the RTT monitor goroutine don't reset the average RTT, so there's no chance of it being reset between the connection handshake and this line.